lib/asset_checker.rb

```diff
@@ -0,0 +1,39 @@
+class AssetChecker
+  def self.run(argv)
+    assets_file = 'app/assets/javascripts/assets.js.erb'
+    translations_file = 'app/assets/javascripts/i18n-strings.js.erb'
+    @asset_strings = load_included_strings(assets_file)
+    @translation_strings = load_included_strings(translations_file)
+    exit_code = (argv.map { |f| check_file(f) }).any? ? 1 : 0
+    exit exit_code # rubocop:disable Rails/Exit
+  end
+
+  def self.check_file(file)
+    data = File.open(file).read
+    missing_found = false
+    missing_translations = find_missing(data, /\Wt\s?\(['"]([^'^"]*)['"]\)/, @translation_strings)
+    missing_assets = find_missing(data, /\WassetPath=["'](.*)['"]/, @asset_strings)
+    if missing_translations.any? || missing_assets.any?
+      missing_found = true
```

Comment:
I would get rid of re-assigning `missing_found` inside the loop, define it once outside, and consider renaming it:

```ruby
has_missing = (missing_translations + missing_assets).any?
```

Reply:
Or potentitally:

```ruby
has_missing = missing_translations.any? || missing_assets.any?

if has_missing
  warn file
  # etc ....
end

has_missing
```